atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/TargetedSweeper.java

```diff
@@ -55,31 +57,33 @@
     private volatile boolean isInitialized = false;
 
     private TargetedSweeper(Supplier<Boolean> runSweep, Supplier<Integer> shardsConfig,
-            int conservativeThreads, int thoroughThreads) {
+            int conservativeThreads, int thoroughThreads, Follower follower) {
         this.runSweep = runSweep;
         this.shardsConfig = shardsConfig;
         this.conservativeScheduler = new BackgroundSweepScheduler(conservativeThreads,
                 TableMetadataPersistence.SweepStrategy.CONSERVATIVE);
         this.thoroughScheduler = new BackgroundSweepScheduler(thoroughThreads,
                 TableMetadataPersistence.SweepStrategy.THOROUGH);
         this.minShards = Math.max(conservativeThreads, thoroughThreads);
+        this.follower = follower;
     }
 
     /**
      * Creates a targeted sweeper, without initializing any of the necessary resources. You must call the
-     * {@link #initialize(SpecialTimestampsSupplier, KeyValueService)} method before any writes can be made to the
-     * sweep queue, or before sweeping.
+     * {@link #initialize(SpecialTimestampsSupplier, KeyValueService, TargetedSweepFollower)} method before any writes
+     * can be made to the sweep queue, or before sweeping.
      *
      * @param enabled live reloadable config controlling whether background threads should perform targeted sweep.
      * @param shardsConfig live reloadable config specifying the desired number of shards. Since the number of shards
      * must never be reduced, this will be ignored if the persisted number of shards is greater.
      * @param conservativeThreads number of conservative threads to use for background targeted sweep.
      * @param thoroughThreads number of thorough threads to use for background targeted sweep.
+     * @param follower follower used for sweeps.
      * @return returns an uninitialized targeted sweeper.
      */
     public static TargetedSweeper createUninitialized(Supplier<Boolean> enabled, Supplier<Integer> shardsConfig,
```

Comment:
I suspect that in the near future we will need support for either multiple followers or to compose followers together.

Reply:
I made it into a list, we can refactor later if needed